aws/cloudformation/moz-single.json

```diff
@@ -274,35 +284,93 @@
       }
     },
 
-    "FxaELB":{
-      "Type":"AWS::ElasticLoadBalancing::LoadBalancer",
+    "FxaLB":{
+      "Type":"AWS::ElasticLoadBalancingV2::LoadBalancer",
       "Properties":{
-        "AvailabilityZones" : { "Fn::GetAZs" : "" },
-        "ConnectionDrainingPolicy":{ "Enabled" : true, "Timeout" : 15 },
-        "HealthCheck":{
-          "Target":"TCP:80",
-          "HealthyThreshold": "2",
-          "UnhealthyThreshold": "2",
-          "Interval": "15",
-          "Timeout": "3"
-        },
-        "Listeners":[
-          {
-            "InstancePort":"1443",
-            "LoadBalancerPort":"443",
-            "Protocol":"HTTPS",
-            "SSLCertificateId":{
-               "Ref" : "SSLCertificateArn"
-            }
-          },
-          {
-            "InstancePort":"80",
-            "LoadBalancerPort":"80",
-            "Protocol":"HTTP"
+        "Subnets": {"Fn::Split": [",", {"Ref": "Subnets"}]},
+        "SecurityGroups": [{ "Fn::GetAtt" : [ "FxaDevLBSecurityGroup", "GroupId" ] }],
+        "LoadBalancerAttributes": [
+          { "Key": "idle_timeout.timeout_seconds", "Value": "500" }
+        ]
+      }
+    },
+
+    "FxaLBTargetPort80":{
+      "Type":"AWS::ElasticLoadBalancingV2::TargetGroup",
+      "Properties":{
+        "TargetType": "instance",
+        "Targets": [{"Id": { "Ref" : "FxaEc2Instance" }, "Port": "80"}],
+        "VpcId": {"Ref": "VPC"},
+        "Port": "80",
+        "Protocol": "HTTP",
+        "HealthCheckProtocol": "HTTP",
+        "HealthCheckIntervalSeconds": "30",
+        "HealthCheckTimeoutSeconds": "3",
+        "HealthyThresholdCount": "2",
+        "UnhealthyThresholdCount": "2",
+        "HealthCheckPath": "/",
+        "Matcher" : {
+          "HttpCode" : "200,301"
```

Comment:
If we decide we like the ALB, we could actually configure this listener to redirect HTTP to HTTPS without hitting the instance at all, which might simplify some of our nginx config.

Reply:
👍